src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2300,6 +2310,19 @@ Object toStyleConstants(StyleConstants key, View v) {
             return family;
         }
 
+        @Override
+        public int hashCode() {
+            if (family != null) {
+                return family.hashCode();
+            } else {
+                return 0;
+            }
+        }
+        @Override
+        public boolean equals(Object val) {
+            return val instanceof CSS.FontFamily font && family == font.family;
```

Comment:
Is the `family` field interned somewhere? If it isn't, then should this be:

```
return val instanceof CSS.FontFamily font && 
    Objects.equals(family, font.family);
```

For ex if we tweak the start of the FontFamily unit test as follows does it still pass?
```
private static void testFontFamily() {
    StyleSheet ss = new StyleSheet();

    SimpleAttributeSet a = new SimpleAttributeSet();
    ss.addCSSAttribute( a, CSS.Attribute.FONT_FAMILY, "Sans-Serif");

    SimpleAttributeSet b = new SimpleAttributeSet();
    ss.addCSSAttribute( b, CSS.Attribute.FONT_FAMILY, "Sans-Serif");
```

Reply:
I'm pretty sure the test will still pass because Java compiler interns strings. You have to explicitly create a new instance of the String to make the test fail.

In real applications, attributes are parsed from a style sheet and therefore the identity test won't pass.

Reply:
yes this needs to be changed.
Also needs to be changed at line 2034.

Reply:
String comparison changed.
Also, I am not able to see this test fail in latest iteration..